Client/src/modeloClases/Batalla.java

```diff
@@ -8,48 +8,94 @@
-                return entrenadorA;
-            }else{
-                if(randomCartaB)
-                    entrenadorA.usarCarta(p1);
-                p2.ataca(p1);
-                if (p1.getVitalidad() == 0){
-                    p2.actualizaClasificacion();
-                    entrenadorB.actualizaClasificacion();
-                    return entrenadorB;
-                }
-            }
+
+        //Me parece que hay que hacer el finally, entonces... quedaria mejor hacer una funcion analizaCartas(..) que se encargue de asignar la carta o tirar la excepcion
+        //Y aca solo poner try{analizaCartas(entrenadorA);  analizaCartas(entrenadorB)} catch { e.getMessage()} finally {etc}
+        //La funcion estaria en esta misma clase, no se si les parece bien la idea. Creeria que tiene que funcionar
+        try {
+        	if (entrenadorA.getCartasDisponibles() > 0) 
+        		randomCartaA = new Random().nextBoolean();
+        	else
+        		throw new SinCartasDisponiblesException("El entrenador "+entrenadorA.getNombre()+" quizo utilizar una carta y no tiene disponibles.\n");
+        }
+        catch(SinCartasDisponiblesException e) {
+        	e.getMessage();
+        }
+        
+        try {
+        	if (entrenadorB.getCartasDisponibles() > 0)
+        		randomCartaB = new Random().nextBoolean();
+        	else
+        		throw new SinCartasDisponiblesException("El entrenador "+entrenadorB.getNombre()+" quizo utilizar una carta y no tiene disponibles.\n");
+        }
+        catch(SinCartasDisponiblesException e) {
+        	e.getMessage();
         }
-        if(p1.comparaEstado(p2)== 1){
-            p1.actualizaClasificacion();
-            entrenadorA.actualizaClasificacion();
-            return entrenadorA;
-        }else{
-            p2.actualizaClasificacion();
-            entrenadorB.actualizaClasificacion();
-             return entrenadorB;
+        
+        while (i<maxAtaques) {
+        	if (randomCartaA) {
+        		entrenadorA.usarCarta(p2);
+        		randomCartaA=false;
+        	}
+        	p1.ataca(p2);
+        	if (p2.getVitalidad()==0) {
+        		p1.actualizaClasificacion();
+        		entrenadorA.actualizaClasificacion();
+        		ganador = entrenadorA;
+        		i=99;
+        	}
+        	else {
+        		if (randomCartaB) {
+        			entrenadorA.usarCarta(p1);
+        			randomCartaB=false;
+        		}
+        		p2.ataca(p1);
+        		if (p1.getVitalidad() == 0) {
+        			p2.actualizaClasificacion();
+        			entrenadorB.actualizaClasificacion();
+        			ganador= entrenadorB;
+        			i=99;
+        		}
+        	}
         }
+        
+        if (ganador==null) {
+        	if (p1.comparaEstado(p2)== 1) {
+        		p1.actualizaClasificacion();
+        		entrenadorA.actualizaClasificacion();
+        		ganador = entrenadorA;
+        	}
+        	else {
+        		p1.actualizaClasificacion();
+        		entrenadorA.actualizaClasificacion();
```

Comment:
```suggestion
        		entrenadorB.actualizaClasificacion();
```

esta mal el entrenador sobre el que se le esta haciendo la actualizacion